src/main/java/org/wildfly/security/auth/server/HttpAuthenticationFactory.java

```diff
@@ -49,7 +61,62 @@ HttpServerAuthenticationMechanism doCreate(final String name, final CallbackHand
     }
 
     Collection<String> getAllSupportedMechNames() {
-        return Arrays.asList(mechanismFactory.getMechanismNames(Collections.emptyMap()));
+        return asList(mechanismFactory.getMechanismNames(Collections.emptyMap()));
+    }
+
+    // TODO: at some point these should no longer be hard-coded
+
+    Collection<Class<? extends Evidence>> getSupportedEvidenceTypes(final String mechName) {
+        switch (mechName) {
+            case HttpConstants.BASIC_NAME: {
+                return singleton(PasswordGuessEvidence.class);
+            }
+            default: {
+                return emptySet();
+            }
+        }
+    }
+
+    Collection<String> getSupportedEvidenceAlgorithmNames(final Class<? extends AlgorithmEvidence> evidenceType, final String mechName) {
+        return emptySet();
+    }
+
+    Collection<Class<? extends Credential>> getSupportedCredentialTypes(final String mechName) {
+        switch (mechName) {
+            case HttpConstants.BASIC_NAME:
+            case "DIGEST": {
+                return singleton(PasswordCredential.class);
```

Comment:
This does not seem quite enough, PasswordCredential could be supported for many password types but it is only a subset of those types that will actually be usable with Digest.